_source/_posts/2019-12-27-spring-boot-deploy-docker.adoc

```diff
@@ -0,0 +1,257 @@
+---
+layout: blog_post
+title: "Add Docker to Your Spring Boot Application"
+author: thiago-negri
+description: "Tutorial: Learn how to package a Spring Boot application as a Docker container."
+tags: [spring-boot, oauth, oidc, docker, deploy]
+tweets:
+- "Create a Docker container with a Spring Boot app with 4 lines of code!"
+- "Learn how to package a @SpringBoot app with @Docker!"
+- "A @SpringBoot app + @Docker == 💕☕"
+image: blog/featured/okta-java-short-tile-books-mouse.jpg
+---
+
+:page-liquid:
+:experimental:
+
+Docker enables you to deploy a server environment in containers. A container is a standardized unit of software that assembles code, runtime, dependencies, settings, and initialization in a single package that you can run reliably from one computing environment to another. Deploying your app as a Docker container is not hard, and can alleviate a lot of problems that you may encounter when moving your app around the multiple environments it has to go, like development machines, QA environment, staging, and production.
+
+image::{% asset_path 'blog/spring-boot-deploy-docker/app-docker-host.png' %}[alt="Docker host layer diagram",width=400,align=center]
+
+In this tutorial, you'll see how to leverage Docker to deploy your Spring Boot application.
+
+**Prerequisites:**
+
+* Java 8+ - Install with https://sdkman.io/SDKMAN or directly from https://adoptopenjdk.net/[AdoptOpenJDK].
+* Docker - I recommend https://www.docker.com/products/docker-desktop[Docker Desktop].
+* An https://developer.okta.com/signup[Okta account] - More on that below.
```

Comment:
I don't think periods are necessary on these bullets. It's OK to leave them though since they are consistent.